src/common/CMakeLists.txt

```diff
@@ -1,8 +1,22 @@
+
+set (MONGOC_ENABLE_TESTING 0)
+
+if (ENABLE_TESTING)
+    set (MONGOC_ENABLE_TESTING 1)
+    unset (ENABLE_TESTING CACHE)
```

Comment:
Though I am not a fan of cmake caching, since `ENABLE_TRACING` is cached, I'd argue for removing this for consistency. It seems like it'd be less of a surprise if this behaved the same way as the other options.